src/Fantomas.Core.Tests/CommentTests.fs

```diff
@@ -2481,3 +2481,81 @@ The same indentation is ensured to be consistent in a source file.
 Default = 4.
 *)
 """
+
+// TODO: drop the sarcasm and give proper names
+// Consider adding a separate file for this, as it does cover a focused problem space
+
+[<Test>]
+let ``the one million dollar fix`` () =
+    formatSourceString
+        false
+        """
+[
+    "thing"
+    "thing2"
+    // no other things
+]
+"""
+        { config with MultilineBlockBracketsOnSameColumn = true }
+    |> prepend newline
+    |> should
+        equal
+        """
+[
+    "thing"
+    "thing2"
+    // no other things
+]
+"""
+
+[<Test>]
+let ``this even works nested, oh my`` () =
+    formatSourceString
+        false
+        """
+let v =
+    [
+        "thing"
+        "thing2"
+        // no other things
+    ]
+"""
+        { config with MultilineBlockBracketsOnSameColumn = true }
+    |> prepend newline
+    |> should
+        equal
+        """
+let v =
+    [
+        "thing"
+        "thing2"
+        // no other things
+    ]
+"""
+
+[<Test>]
+let ``how adding one blank line really complicates things`` () =
+    formatSourceString
+        false
+        """
+let v =
+    [|
+        "thing"
+        "thing2"
+
+        // there is a blank line above this comment
```

Comment:
This required the `Newline` and `CommentOnSingleLine` to be assigned together to a fitting node.
That group requires knowledge of the start column of the comment in this case.
I don't think the grouping concept is a bad one, it does ripple through a lot of the existing code. I think some more aspects will need to be revisited before this can be finalized.